### PR TITLE
May need more work?

### PR DESCRIPTION
**Pull request by ixnp**

### CODE REVIEW

File: bus-mall.html [hunk 1]
```diff
@@ -0,0 +1,41 @@
+<!DOCTYPE html>
+<html>
+<head>
+<title>Bus-mall</title>
+    <link rel="stylesheet" type="text/css" href="bus-mall.css">
+</title>
+<body>
+  <header>
+    <secton class="logo-info">
+      <img src="" alt="">
```
Comment (fraziermork) [inline bus-mall.html:10]: What is the purpose of this img element?

File: bus-mall.html [hunk 2]
```diff
@@ -0,0 +1,41 @@
+<!DOCTYPE html>
+<html>
+<head>
+<title>Bus-mall</title>
+    <link rel="stylesheet" type="text/css" href="bus-mall.css">
+</title>
+<body>
+  <header>
+    <secton class="logo-info">
```
Comment (fraziermork) [inline bus-mall.html:9]: typo!

File: bus-mall.html [hunk 3]
```diff
@@ -0,0 +1,41 @@
+<!DOCTYPE html>
+<html>
+<head>
+<title>Bus-mall</title>
+    <link rel="stylesheet" type="text/css" href="bus-mall.css">
```
Comment (fraziermork) [inline bus-mall.html:5]: Pay attention to your indentation!

File: bus-mall.html [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+      <ul>
+          <li>Home</li>
+          <li>Products</li>
+          <li>About</li>
+          <li>Contact</li>
+      </ul>
+   </nav>
+  </header>
+  <main>
+    <aside id= "results"></aside>
```
Comment (fraziermork) [inline bus-mall.html:23]: No spaces next to equals signs in html.

File: bus-mall.html [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+        <button type="button" Id="show-results">Results</button>
+        <p>ldasjgfadgjnanjdj</p>
+    </section>
+
+</main>
+<script src="bus-mall.js"></script>
+</body>
+
+
+<!-- </html> -->
```
Comment (fraziermork) [inline bus-mall.html:41]: Why did you comment this out?

File: bus-mall.html [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+        <img  id="imageThree">
+      </section>
+    </section>
+    <section>
+        <button type="button" Id="show-results">Results</button>
+        <p>ldasjgfadgjnanjdj</p>
+    </section>
+
+</main>
+<script src="bus-mall.js"></script>
```
Comment (fraziermork) [inline bus-mall.html:37]: Make sure to be attentive to your indentation!

File: bus-mall.js [hunk 7]
```diff
@@ -0,0 +1,109 @@
+// refresh page, .random choseing imgags, images stored in array, <15, create counter/
+//whileloop. event listner < on click refresh page
+
+
+var productNames = ['bag', 'banana', 'bathroom', 'boots', 'breakfast', 'bubblegum', 'chair', 'cthulhu', 'dog-duck', 'dragon', 'pen', 'pet-sweep', 'scissors', 'shark', 'tauntaun', 'unicorn', 'water-can', 'wine-glass'];
+
+var productArr = [];
```
Comment (fraziermork) [inline bus-mall.js:7]: Can you think of a clearer name for this? Maybe allProducts? 

Also, as an aside, it's generally considered a no-no to describe the type of a variable in the name of it, like specifying that it's an array. That tells you what something is, but it gives zero information about what it's for. It's ok while you're learning to code, but it's something to look out for in the future.

File: bus-mall.js [hunk 8]
```diff
@@ -0,0 +1,109 @@
...
+
+// Constructor function
+function Product(name) {
+ this.name = name;
+ this.votes = 0;
+ this.path = 'image/' + name + '.jpg';
+ productArr.push(this);
+}
+console.log(this.path);
```
Comment (fraziermork) [inline bus-mall.js:16]: When does this console log run? Does it run when the Product function is called? Or does it print `undefined` once?

File: bus-mall.js [hunk 9]
```diff
@@ -0,0 +1,109 @@
...
+// a simple IIFE to build all the product images // Producing path to new product
+(function () {
+ for (var i = 0; i < productNames.length; i++) {
+   new Product(productNames[i]);
+ }
+ console.log(productArr);
+})();
+
+var tracker = {
+  imageOneEl: document.getElementById('imageOne'),
```
Comment (fraziermork) [inline bus-mall.js:26]: When you have three identical things that you want to iterate through, what do you put them in?

File: bus-mall.js [hunk 10]
```diff
@@ -0,0 +1,109 @@
...
+  imgObjOne: null,
+  imgObjTwo: null,
+  imgObjThree: null,
+  clicks: 0,
+
+
+getRandomIndex:function(){
+  return Math.floor(Math.random()* productArr.length);
+  console.log(getRandomIndex);
+  },
```
Comment (fraziermork) [inline bus-mall.js:41]: Indentation!

File: bus-mall.js [hunk 11]
```diff
@@ -0,0 +1,109 @@
...
+
+  displayImages: function() {
+    this.imgObjOne = productArr[this.getRandomIndex()];
+    this.imgObjTwo = productArr[this.getRandomIndex()];
+    this.imgObjThree = productArr[this.getRandomIndex()];
+//keeps out dulplicats//
+
+
+    if(this.imgObjOne === this.imgObjTwo || this.imgObjOne === this.imgObjThree || this.imgObjTwo === this.imgObjThree){
```
Comment (fraziermork) [inline bus-mall.js:50]: Can you think of a cleaner way to do a check like this?

File: bus-mall.js [hunk 12]
```diff
@@ -0,0 +1,109 @@
...
+    this.displayImages();
+  }
+
+  this.imageOneEl.src = this.imgObjOne.path;
```
Comment (fraziermork) [inline bus-mall.js:54]: When you have a repeated task like this, what's the traditional way to do it?

File: bus-mall.js [hunk 13]
```diff
@@ -0,0 +1,109 @@
...
+  this.imageTwoEl.src = this.imgObjTwo.path;
+  this.imageTwoEl.id = this.imgObjTwo.name;
+  this.imageThreeEl.src = this.imgObjThree.path;
+  this.imageThreeEl.id = this.imgObjThree.name;
+//29:00//
+  },
+
+ checkClicks: function(){
+   console.log(this.clicks);
+  if(this.clicks > 14){
```
Comment (fraziermork) [inline bus-mall.js:65]: Where does 14 come from?

File: bus-mall.js [hunk 14]
```diff
@@ -0,0 +1,109 @@
...
+      e.target.id === tracker.imgObjOne.name ||
+      e.target.id === tracker.imgObjTwo.name ||
+      e.target.id === tracker.imgObjThree.name) {
+      tracker.clicks++;
+      tracker.tallyVotes(e.target.id);
+      tracker.displayImages();
+    }
+ },
+
+ tallyVotes: function(elId){
```
Comment (fraziermork) [inline bus-mall.js:86]: Nice!

File: bus-mall.js [hunk 15]
```diff
@@ -0,0 +1,109 @@
...
+     liEl.textContent = productArr[i].name + ':' + productArr[i].votes;
+     ulEl.appendChild(liEl);
+   }
+   this.resultsEl.appendChild(ulEl);
+ }
+};
+tracker.imageContainerEl.addEventListener('click', tracker.clickHandler);
+
+
+tracker.displayImages();
```
Comment (fraziermork) [inline bus-mall.js:109]: Great!

**Review by fraziermork — COMMENTED** [review]
Really nice work!

File: bus-mall.html (hunk 1 above)
Comment (ixnp) [reply bus-mall.html:10]: This is for styling later.

File: bus-mall.js (hunk 13 above)
Comment (ixnp) [reply bus-mall.js:65]: I guess this could also be .length but I wasn't sure if the number of clicks was relevant to the number of images.

**Pull request merged**